core/site.go

```diff
@@ -873,6 +921,7 @@ func (site *Site) Run(stopC chan struct{}, interval time.Duration) {
 		select {
 		case <-ticker.C:
 			site.update(<-loadpointChan)
+			site.UpdateBatteryMode(site.Loadpoints())
```

Comment:
Idee: reicht es nicht, den immer nächsten Loadpoint zu betrachten anstatt alle durchzuklappern? Die können sich ja nicht asynchron ändern?

Reply:
Ich finds auch nicht so schön, dass die Logik hier jetzt außerhalb von update() liegt wo sonst alles drin ist.

Reply:
Es müssen immer alle Loadpoints betrachtet werden, da sie nur in der Gesamtheit eine Aussage darüber machen ob die Batterieentladung unterbunden werden soll. Wenn wir nacheinander draufschauen, führt dies dazu, das wir für einen Loadpoint zum Ergebnis kommen das die Batterie entladen darf und beim nächsten Loadpoint aber genau andersrum entscheiden. Dann schalten wir die Batterie ständig an und aus.

Ich hatte ursprünglich vor es in update() einzubauen, hatte dann aber realisiert das update() pro Loadpoint aufgerufen wird und es dann wegen obiger Argumentation bewusst rausgezogen.
Gleichzeitig habe ich aber gesehen, daß am Anfang von update()  ChargePower einmal über alle Loadpoints refreshed wird. Da könnte man argumentieren, daß update() bereits jetzt nicht nur den aktuellen Loadpoint betrachtet und daher es auch als OK ansehen die Batterie Logik dort reinzuziehen.

Reply:
Stimmt :/

> Ich hatte ursprünglich vor es in update() einzubauen, hatte dann aber realisiert das update() pro Loadpoint aufgerufen wird und es dann wegen obiger Argumentation bewusst rausgezogen.

Du rufst es doch jetzt auch bei jedem Ladepunktupdate auf? 

>  Da könnte man argumentieren, daß update() bereits jetzt nicht nur den aktuellen Loadpoint betrachtet und daher es auch als OK ansehen die Batterie Logik dort reinzuziehen.

Oder so. Tatsächlich könnte man sich das ganze Abklappern dann aber auch sparen und einfach auf die Chargepower schauen die man eh schon hat? Dann wiederrum könnte man sich da an einem Schwellwert orientieren?

Reply:
> Du rufst es doch jetzt auch bei jedem Ladepunktupdate auf?

Nein, einmal pro Timer Intervall. Aber jetzt wo ich genau hinschaue - kurz Off Topic. Hier ist generell das Handling der Loadpoints anders als ich erwarten würde. Wenn ich es richtig sehe, wird pro Intervall immer nur 1 nächster Loadpoint aus dem Channel geholt. Bei mehreren Loadpoints werden diese also immer im Wechsel reihum abgefragt. Das "Intervall" für den einzelnen Loadpoint ist dann `<anzahl loadpoints>*<intervall>`. Ist das wirklich so gewünscht?

Back on topic. Ich denke die Logik muß trotzdem ins update() für den einzelnen Loadpoint, da wir ja auch über die UI gezielte Updates auf den Loadpoint bekommen, da wollen wir ja auch die passenden Werte ausrechnen und umschalten.

Das mit dem Schwellwert ist eine spannende Frage. Ich weiß das ThinkEV seine Lösung mit einem Schwellwert versehen hat. Andererseits reicht die reine Chargepower meiner Ansicht nach nicht aus. Es kommt ja auch drauf an, wo diese herkommt. Die kann ja auch bei PV relativ hoch sein - je nach Peakleistung der Anlage. Dann müßte man zumindest die Erzeugungsleistung mit reinnehmen und wenn diese 0 ist den Schwellwert nochmal anschauen. Außerdem gibt es dann vermutlich auch verschiedene Meinungen was der richtige Wert für diese Schwelle ist. Ich finde die jetzige Logik basierend auf Loadpoint Mode und PlannerActive besser, weil einfacher.

Reply:
Ich habe gestern Abend noch den Aufruf von updateBatteryMode() in update() verschoben. Sollen wir dann hier erstmal resolven?